sync2jira/intermediary.py

```diff
@@ -277,33 +278,37 @@ def map_fixVersion(mapping, issue):
         issue["milestone"] = fixVersion_map.replace("XXX", issue["milestone"])
 
 
-def matcher(content, comments):
+JIRA_REFERENCE = re.compile(r"Relates +to +JIRA:\s*(\w+-\d+)")
+
+
+def matcher(content: Optional[str], comments: list[dict[str, str]]) -> str:
     """
     Helper function to match to a JIRA
 
+    Extract the Jira ticket reference from the first instance of the magic
+    cookie (e.g., "Relates to JIRA: FACTORY-1234") found when searching
+    through the comments in reverse order.  If no reference is found in the
+    comments, then look in the PR description.  This ordering allows later
+    comments to override earlier ones as well as any reference in the
+    description.
+
     :param String content: PR description
     :param List comments: Comments
     :return: JIRA match or None
     :rtype: Bool
     """
-    # Build out a string with all comments and initial_comment
-    all_data = " "
+
+    def find_it(input_str: str) -> str:
+        match = JIRA_REFERENCE.search(input_str)
+        return match.group(1) if match else None
+
     for comment in reversed(comments):
-        all_data += f" {comment['body']}"
-    if content:
-        all_data += content
-    if all_data:
-        # Parse to extract the JIRA information. 2 types of matches:
-        # 1 - To match to JIRA issue (i.e. Relates to JIRA: FACTORY-1234)
-        # 2 - To match to upstream issue (i.e. Relates to Issue: !5)
-        match_jira = re.findall(r"Relates to JIRA: ([\w]*-[\d]*)", all_data)
```

Comment:
@ralphbean, @Zyzyx:  the code only matches case 1, so in my change I've removed this comment.  Is the code _supposed_ to work for case 2??  (If so, I can correct the code, instead....)

Reply:
FWIW, this code seems substantially unchanged since it was originally added in #8...so I guess it _never_ worked for case 2 (the `!` wouldn't match the `-`).